nav2_planner/src/planner_server.cpp

```diff
@@ -250,6 +250,11 @@ bool PlannerServer::isServerInactive(
 
 void PlannerServer::waitForCostmap()
 {
+  if (costmap_ros_->isUpdateOnRequest()){
+    RCLCPP_DEBUG(get_logger(), "Updating costmap on request!");
+    costmap_ros_->updateMapOnRequest();
+    return;
```

Comment:
Why return should be here?

Reply:
My mistake. The return should not be there because the function should still continue to wait until the costmap has been fully updated